test/jdk/javax/print/attribute/OutputBinAttributePrintDialogTest.java

```diff
@@ -0,0 +1,323 @@
+            for (OutputBin outputBin : supportedOutputBins) {
+                testPrint(dialogTypeSelection, outputBin, supportedOutputBins);
+            }
+            testFinished = true;
+        });
+
+        long time = System.currentTimeMillis() + TIMEOUT;
+
+        while (System.currentTimeMillis() < time) {
+            if (!testPassed || testFinished) {
+                break;
+            }
+            Thread.sleep(500);
+        }
+
+        timeout = true;
+
+        closeDialogs();
+
+        if (!testPassed) {
+            throw new Exception("Test failed!");
+        }
+
+        if (testCount != testTotalCount) {
+            throw new Exception(
+                    "Timeout: " + testCount + " tests passed out from " + testTotalCount);
+        }
+    }
+
+    private static void print(DialogTypeSelection dialogTypeSelection, OutputBin outputBin) throws PrinterException {
+        PrintRequestAttributeSet attr = new HashPrintRequestAttributeSet();
+        attr.add(MediaSizeName.ISO_A4);
+        attr.add(dialogTypeSelection);
+
+        for (Attribute attribute : attr.toArray()) {
+            System.out.printf("Used print request attribute: %s%n", attribute);
+        }
+
+        PrinterJob job = PrinterJob.getPrinterJob();
+        job.setJobName("Print to " + outputBin + " output bin through " + dialogTypeSelection + " print dialog");
+        job.setPrintable(new OutputBinAttributePrintable(outputBin));
+
+        if (job.printDialog(attr)) {
+            job.print();
+        } else {
+            throw new RuntimeException(dialogTypeSelection + " print dialog for " + outputBin + " is canceled!");
+        }
+    }
+
+    private static class OutputBinAttributePrintable implements Printable {
+
+        private final OutputBin outputBinAttr;
+
+        public OutputBinAttributePrintable(OutputBin outputBinAttr) {
+            this.outputBinAttr = outputBinAttr;
+        }
+
+        @Override
+        public int print(Graphics graphics, PageFormat pageFormat, int pageIndex) throws PrinterException {
+
+            if (pageIndex != 0) {
+                return NO_SUCH_PAGE;
+            }
+
+            int x = (int) (pageFormat.getImageableX() + pageFormat.getImageableWidth() / 10);
+            int y = (int) (pageFormat.getImageableY() + pageFormat.getImageableHeight() / 5);
+
+            Graphics2D g = (Graphics2D) graphics;
+            g.setColor(Color.BLACK);
+            g.drawString(getPageText(outputBinAttr), x, y);
+            return PAGE_EXISTS;
+        }
+    }
+
+    private static String getPageText(OutputBin outputBin) {
+        return String.format("Output bin: %s", outputBin);
+    }
+
+    private static OutputBin[] getSupportedOutputBinttributes() {
+
+        PrinterJob printerJob = PrinterJob.getPrinterJob();
+        PrintService service = printerJob.getPrintService();
+        if (service == null) {
+            return new OutputBin[0];
+        }
+
```

Comment:
I guess before getting supported Attribute Values, we need to ensure category is supported by calling `service.isAttributeCategorySupported(OutputBin.class)`
in my opinion

Reply:
The test updated to check if the OutputBin category is supported.

Reply:
I think if `isATtributeCategorySupported `returns false, we should not return OutputBin[0] as that will cause the test to pass with message "SKip the test as number of supported outputbins is less than 2", which is misleading, in my opinion
It should instead message "Skipping the test as OutputBin category is not supported for this printer"

Also, `getPrintService` returning null also return the above outputbin message..It should rather return "No print service found"

Reply:
The test is updated to print messages if `getPrintService()` returns null or the OutputBin category is not supported and check the case that when the OutputBin category is supported there should be supported output bins.